addons/mail/static/src/models/thread/thread.js

```diff
@@ -1400,6 +1400,17 @@ function factory(dependencies) {
             );
         }
 
+        /**
+         * Compute an url string that can be used inside a href attribute
+         *
+         * @private
+         * @returns {string}
+         */
+        _computeUrl() {
+            const baseHREF = this.env.session.url('/web');
+            return `${baseHREF}#model=${this.model}&id=${this.id}`;
+        }
```

Comment:
Looks correct for non-`mail.channel` threads (which are coupled with related record form sheet).

For `mail.channel`, I would expect it to open discuss with the appropriate channel. Can make use of `active_id` with format `mail.channel_<id>`.